apps/api/src/common/config/index.ts

```diff
@@ -54,6 +20,25 @@ if (!env.success) {
 
 const { data: envData } = env;
 
+const createMetadataConfig = (env: typeof envData): MetadataConfig<MetadataSource> => {
+    switch (env.METADATA_SOURCE) {
```

Comment:
is this returning the types we expect ? 
Which type is this function returning if we can return any of the metadata sources ?

Reply:
as we discussed, i found a better strongly typed way from zod to make the validation of metadata based on each so this type is inferred correctly here